webapp/lib/authentication.dart

```diff
@@ -103,7 +103,7 @@ void authStateObserver(firebase.User user) {
     String datasetName = Uri.base.queryParameters["dataset"];
     var dataset;
     try {
-      dataset = fbt.loadDataset(datasetName);
+      dataset = await fbt.loadDataset(datasetName);
     } catch (e) {
       ui.codaUI.displayUrlErrorView(e.toString());
```

Comment:
Can you confirm that you've tested that this exception handler still works correctly - async await does very strange things to exceptions

Reply:
Yep, this seems to be working fine, thanks for flagging it though.